File: app/src/main/java/com/example/task2/javafolder/viewviewmodel/fragments/parentfragment/maps/MapsFragment.java
package com.example.task2.javafolder.viewviewmodel.fragments.parentfragment.maps;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.example.task2.javafolder.viewviewmodel.fragments.parentfragment.ParentFragment;
import com.example.task2.javafolder.viewviewmodel.fragments.parentfragment.ParentViewModel;

public class MapsFragment extends ParentFragment {
    MapsViewModel viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this).get(MapsViewModel.class);
    }

    @Override
    public int getSpanCount() {
        return 2;
    }

    @Override
    public ParentViewModel determineViewModelProvider() {
        return viewModel;
    }
}

File: app/src/main/java/com/example/task2/kotlinfolder/viewviewmodel/fragments/parentfragment/maps/MapsFragment.kt
package com.example.task2.kotlinfolder.viewviewmodel.fragments.parentfragment.maps

import com.example.task2.kotlinfolder.viewviewmodel.fragments.BenchmarkAdapter
import com.example.task2.kotlinfolder.viewviewmodel.fragments.parentfragment.ParentFragment
import com.example.task2.kotlinfolder.viewviewmodel.fragments.parentfragment.ParentViewModel
import org.koin.androidx.viewmodel.ext.android.viewModel

class MapsFragment : ParentFragment<MapsViewModel>(BenchmarkAdapter()) {

    override val viewModel: MapsViewModel by viewModel()

    override val spanCount = 2
}